components/bl-navigation-dots/src/index.js

```diff
@@ -0,0 +1,167 @@
+
+    const activeAnchor = anchors[index];
+    const sectionElement = document.getElementById(activeAnchor);
+
+    if (sectionElement) {
+      sectionElement.scrollIntoView({ behavior: scrollBehavior });
+      onAnchorChange({ activeAnchor });
+    }
+
+    setTimeout(() => setScrollEnabled(true), 1000);
+  }, [anchors, onAnchorChange, smoothScroll]);
+
+  useEffect(() => {
+    document.documentElement.style.scrollBehavior = scrollBehavior;
+  }, [smoothScroll]);
+
+  useEffect(() => {
+    window.addEventListener('scroll', scrollHandler);
+
+    return () => window.removeEventListener('scroll', scrollHandler);
+  }, [scrollHandler]);
+
+  useEffect(() => {
+    anchorValidation(anchors);
+  }, [anchors]);
+
+  component.getActiveAnchor = () => anchors[activeDotIndex];
+  component.setActiveAnchor = anchor => {
+    const index = anchors.findIndex(a => a === anchor);
+
+    if (index !== -1) {
+      dotClickHandler(index);
+    }
+  };
+
+  if (!display) {
+    return null;
+  }
+
+  return (
+    <div className={ cn('bl-customComponent-navigation-dots', classList) } ref={ elRef } style={ styles }>
+
+      { anchors &&
+        <Nav
+          anchors={ anchors }
+          activeDotIndex={ activeDotIndex }
+          dotShape={ dotShape }
+          tooltip={ tooltip }
+          dotClickHandler={ dotClickHandler }
+        />
+      }
+
+    </div>
+  );
+}
+
+function Nav({ anchors, activeDotIndex, dotShape, tooltip, dotClickHandler }) {
+  return (
+    <ul className="nav">
+
+      { anchors.map((anchor, index) => (
+        <NavDotItem
+          key={ anchor }
+          index={ index }
+          isActive={ activeDotIndex === index }
+          dotShape={ dotShape }
+          anchor={ anchor }
+          tooltip={ tooltip }
+          onClick={ dotClickHandler }
+        />
+      )) }
+
+    </ul>
+  );
+}
+
+function NavDotItem({ index, isActive, dotShape, anchor, tooltip, onClick }) {
+  return (
+    <li>
+      <a
+        className={ cn('nav-item', dotShape, { active: isActive }) }
+        href={ `#${ anchor }` }
+        onClick={ () => onClick(index) }>
+
+        { tooltip && <span className="nav-item-tooltip">{ anchor }</span> }
+
+      </a>
+    </li>
+  );
+}
+
+function calculateActiveDotIndex(anchors) {
+  const windowHeight = window.innerHeight;
+
+  let newActiveDotIndex = -1;
+
+  for (let i = 0; i < anchors.length; i++) {
+    const sectionElement = document.getElementById(anchors[i]);
+
+    if (sectionElement) {
+      const rect = sectionElement.getBoundingClientRect();
+
+      if (rect.top <= windowHeight / 2 && rect.bottom >= windowHeight / 2) {
+        newActiveDotIndex = i;
+        break;
+      }
+    }
+  }
+
+  return newActiveDotIndex;
+}
+
+function anchorValidation(anchors) {
+  if (anchors && !anchors.every(anchor => anchor && typeof anchor === 'string')) {
+      console.error("Error: anchor in 'anchors' prop is missing, invalid, or contains non-string value in NavigationDotsComponent.");
```

Comment:
Not quite clear which anchor is missing. How about validate them one by one, and if some is invalid - print it with index in array?